salt/utils/json.py

```diff
@@ -27,10 +27,48 @@ def __split(raw):
 def find_json(raw):
     """
     Pass in a raw string and load the json when it starts. This allows for a
-    string to start with garbage and end with json but be cleanly loaded
+    string to start or end with garbage but the JSON be cleanly loaded
     """
     ret = {}
     lines = __split(raw)
+    lengths = list(map(len, lines))
+    starts = []
+    ends = []
+
+    # Search for possible starts and ends of the json fragments
+    for ind, _ in enumerate(lines):
+        line = lines[ind].lstrip()
```

Comment:
> it can be just for ind, line in enumerate(lines): 

Sure, implemented.

> The end character does not have to be the first character.

This is incorrect. The whole use-case I am adding is for the valid JSON to contain text after the final closing bracket. The function, previously, expects the final bracket to be on a single line by itself, i.e.:

```
Text before json unrelated to the json
{
... json body ...
}
Text after json unrelated to the json
```

The current implementation deals with this:

```
Text before json unrelated to the json
{
...json body...
} text after json unrelated to the json itself
```

We are not adding support for unformatted JSONs, nor does the original implementation support that. Rejecting.